sanity-io/sdk
# chore(core): add Zustand and example implementation

### Description

This PR [Zustand](https://zustand.docs.pmnd.rs/getting-started/introduction)  and a main zustand store that [slices](https://zustand.docs.pmnd.rs/guides/slices-pattern) in a schemaStore. These are used mainly as an example implementation.

This also adds the [devTools middleware](https://zustand.docs.pmnd.rs/middlewares/devtools) so that the[ Redux devtools extension](https://chromewebstore.google.com/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) can be used to inspect the store and view actions.

![CleanShot 2024-11-05 at 15 27 27@2x](https://github.com/user-attachments/assets/4ff07f63-8664-4eeb-8ec6-ecb540ae6348)

![CleanShot 2024-11-06 at 13 12 20 2](https://github.com/user-attachments/assets/210116d6-92b5-4f40-85e1-01eb06eca690)

### What to review
Check out the new Zustand stuff


### Testing

1. Run `pnpm dev:kitchensink`
2. See that the schema is initialized
3. Click the "set schema" button and see that the schema is updated
4. Open the dev tools Redux extension and see the Zustand store and actions that have been fired

### Fun gif
![zustand bear](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExbnMxZWxjbHFzNWdsMDV2NnNmMnRqdjdxdmF2a2F0bDBybDhycHUzYSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/dzaUX7CAG0Ihi/giphy.gif)
